ts/Accessibility/KeyboardNavigation.ts

```diff
@@ -445,8 +446,23 @@ class KeyboardNavigation {
      * @private
      */
     private removeExitAnchor(): void {
-        if (this.exitAnchor && this.exitAnchor.parentNode) {
-            this.exitAnchor.parentNode.removeChild(this.exitAnchor);
+        // Remove event from element and from eventRemovers array to prevent
+        // memory leak (#20329).
+        if (this.exitAnchor) {
+            if (defined((this.exitAnchor as any).focusEventRemover)) {
+                const i =
+                this.eventProvider.eventRemovers.indexOf(
+                    (this.exitAnchor as any).focusEventRemover);
+
+                this.eventProvider.eventRemovers[i]();
```

Comment:
We should just add an `eventProvider.removeEvent()` function that handles this I think, like we have `domElementProvider.removeElement()`.